src/components/Form/Widgets/FileWidget.tsx

```diff
@@ -0,0 +1,216 @@
+import UploadFileIcon from '@mui/icons-material/UploadFile';
+import FilePresentIcon from '@mui/icons-material/FilePresent';
+import * as React from 'react';
+import { type DropzoneOptions, useDropzone } from 'react-dropzone';
+import {
+	Box,
+	Link,
+	Typography,
+	CircularProgress,
+	Avatar,
+	InputLabel,
+	List,
+	ListItem,
+	ListItemText,
+} from '@mui/material';
+import type { WidgetProps } from '@rjsf/utils';
+
+export interface OnFileReadSuccessParams {
+	dataUrl: string;
+	uploadedFile: File;
+}
+
+type FileWidgetProps = WidgetProps & {
+	onChange: (params?: OnFileReadSuccessParams) => void;
+	onError?: (errors: string[]) => void;
+	value?: OnFileReadSuccessParams | string;
+};
+
+export const FileWidget = ({
+	onChange,
+	onError,
+	value,
+	...props
+}: FileWidgetProps) => {
+	const [isLoading, setIsLoading] = React.useState<boolean>(false);
```

Comment:
FYI the `<boolean>` should not be necessary, the state should know its type because you passed a boolean for its value

Reply:
Same for the `<number>` below